### DISCUSSION TITLE
Uvicorn does not work with profiler in pycharm

### DISCUSSION THREAD

**Discussion by ryanisn** [Kludex/uvicorn#2283] [Potential Issue]
This issue has been reported by many people in this thread https://youtrack.jetbrains.com/issue/PY-50507
I'm not confident to say which party should fix this issue, but it does look like `import logging` in a `logging.py` would create circular import under certain circumstance
https://github.com/encode/uvicorn/blob/a2219eb2ed2bbda4143a0fb18c4b0578881b1ae8/uvicorn/logging.py#L4

**Comment by jmgomezsoriano** [comment]
Instead of executing the service with the command uvicorn, you need add this code at the end of the API:


```python
if __name__ == '__main__':
    import uvicorn
    uvicorn.run(app)
```

And execute as normal py program while your do the profile.
